From 09afaaac09c2fc146787d2429c15b89d8db5cc77 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Wed, 3 May 2017 05:16:29 -0400
Subject: [PATCH] fix configure and compile failures

1. Fix do_configure failure
---------------------------
|Checking header volume_key/libvolume_key.h existence and usability.
../tmp/6tvtK.c:1:38: fatal error: volume_key/libvolume_key.h:
No such file or directory
| #include <volume_key/libvolume_key.h>
|Checking header dmraid/dmraid.h existence and usability.../tmp/
ktVJ6.c:1:27: fatal error: dmraid/dmraid.h: No such file or directory
| #include <dmraid/dmraid.h>
---------------------------
We explictly add volume_key and dmraid to DEPENDS, do not need
configure to test.

2. Fix do_compile failure:
---------------------------
|../../../git/src/plugins/crypto.c:24:27: fatal error:
libvolume_key.h: No such file or directory
| #include <libvolume_key.h>
                           ^
---------------------------

Use `-I${STAGING_INCDIR}/volume_key' to replace hardcoded
`-I/usr/include/volume_key'

3. Fix config.h not found
Add it to configure.ac

4. Use pkg-config of dmraid to replace `-ldmarid' for the
generation of gobject-introspection

Upstream-Status: Inappropriate [wr-installer specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 configure.ac            | 4 ++--
 src/lib/Makefile.am     | 4 ++--
 src/plugins/Makefile.am | 2 +-
 3 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/configure.ac b/configure.ac
index ded6425..120ff78 100644
--- a/configure.ac
+++ b/configure.ac
@@ -9,6 +9,8 @@ AC_DISABLE_STATIC
 AM_INIT_AUTOMAKE([foreign -Wall -Werror -Wno-syntax -Wno-portability])
 AC_CONFIG_MACRO_DIR([m4])
 
+AC_CONFIG_HEADERS([config.h])
+
 AM_PATH_PYTHON
 
 AM_PROG_AR
@@ -128,7 +130,6 @@ LIBBLOCKDEV_PKG_CHECK_MODULES([GIO], [gio-2.0 >= 2.42.2])
 AS_IF([test "x$with_crypto" != "xno"],
       [LIBBLOCKDEV_PKG_CHECK_MODULES([CRYPTSETUP], [libcryptsetup >= 1.6.7])
        LIBBLOCKDEV_PKG_CHECK_MODULES([NSS], [nss >= 3.18.0])
-       LIBBLOCKDEV_CHECK_HEADER([volume_key/libvolume_key.h], [$GLIB_CFLAGS $NSS_CFLAGS], [libvolume_key.h not available])
       ],
       [])
 
@@ -138,7 +139,6 @@ AS_IF([test "x$with_dm" != "xno" -o "x$with_lvm" != "xno" -o "x$with_lvm_dbus" !
 
 AS_IF([test "x$with_dm" != "xno"],
       [LIBBLOCKDEV_PKG_CHECK_MODULES([UDEV], [libudev >= 216])
-       LIBBLOCKDEV_CHECK_HEADER([dmraid/dmraid.h], [], [dmraid.h not available])
       ],
       [])
 
diff --git a/src/lib/Makefile.am b/src/lib/Makefile.am
index da16010..56cf357 100644
--- a/src/lib/Makefile.am
+++ b/src/lib/Makefile.am
@@ -34,9 +34,9 @@ BlockDev-2.0.gir: libblockdev.la
 BlockDev_2_0_gir_FILES = $(GIHEADERS)
 BlockDev_2_0_gir_LIBS = libblockdev.la
 BlockDev_2_0_gir_INCLUDES = GObject-2.0
-BlockDev_2_0_gir_PACKAGES = devmapper libudev libcryptsetup libparted blkid bytesize
+BlockDev_2_0_gir_PACKAGES = devmapper libudev libcryptsetup libparted blkid bytesize dmraid
 BlockDev_2_0_gir_CFLAGS = -I${builddir}/../../include/
-BlockDev_2_0_gir_LDFLAGS = -lm -ldmraid -L${builddir}/../utils/ -lbd_utils
+BlockDev_2_0_gir_LDFLAGS = -L${builddir}/../utils/ -lbd_utils
 BlockDev_2_0_gir_SCANNERFLAGS = --warn-error --warn-all --identifier-prefix=BD --symbol-prefix=bd
 BlockDev_2_0_gir_EXPORT_PACKAGES = blockdev
 
diff --git a/src/plugins/Makefile.am b/src/plugins/Makefile.am
index 78c8815..e1315de 100644
--- a/src/plugins/Makefile.am
+++ b/src/plugins/Makefile.am
@@ -70,7 +70,7 @@ if WITH_CRYPTO
 libbd_crypto_la_CFLAGS = $(GLIB_CFLAGS) $(CRYPTSETUP_CFLAGS) $(NSS_CFLAGS) -Wall -Wextra -Werror
 libbd_crypto_la_LIBADD = $(GLIB_LIBS) $(CRYPTSETUP_LIBS) $(NSS_LIBS) -lvolume_key ${builddir}/../utils/libbd_utils.la
 libbd_crypto_la_LDFLAGS = -L${srcdir}/../utils/ -version-info 2:0:0 -Wl,--no-undefined
-libbd_crypto_la_CPPFLAGS = -I${builddir}/../../include/ -I/usr/include/volume_key
+libbd_crypto_la_CPPFLAGS = -I${builddir}/../../include/ -I${STAGING_INCDIR}/volume_key
 libbd_crypto_la_SOURCES = crypto.c crypto.h
 endif
 
-- 
2.8.1

